ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/ActionManager.java

```diff
@@ -39,9 +39,31 @@ public void remove(Uid id)
 		_allActions.remove(id);
 	}
 
-    public int getNumberOfInflightTransactions()
+	/**
+	 * <p>
+	 *     This method works out the number of in-flight transactions currently in
+	 *     the system.
+	 * </p>
+	 * <p>
+	 *     Note: in-flight transactions are defined as all those transactions that
+	 *     haven't started the second phase of the two-phase commit (2PC) protocol.
+	 *     The definition of in-flight (AKA active) transaction, can be found in
+	 *     the OMG OTS specification: [active is defined as] the state of a
+	 *     transaction when processing is in progress and completion of the
+	 *     transaction has not yet commenced.
+	 * </p>
+	 *
+	 * @return the number of in-flight transactions currently in the system
+	 */
+	  public int getNumberOfInflightTransactions()
     {
-        return _allActions.size();
+        return (int) _allActions.entrySet()
```

Comment:
I'm including an earlier comment here: I think == ActionStatus.RUNNING is enough to pick out an in flight transaction. Even though an ABORT_ONLY transaction has not started the 2PC protocol it is not considered to be active.

Reply:
Done. Thanks Michael